server/test/unit/server/async/test_app.py

```diff
@@ -16,17 +23,36 @@ class InitializeWorkerTestCase(unittest.TestCase):
     @mock.patch('pulp.server.async.app.common_utils.create_worker_working_directory')
     @mock.patch('pulp.server.async.app.initialization.initialize')
     @mock.patch('pulp.server.async.app.tasks._delete_worker')
-    def test_initialize_worker(self, _delete_worker, initialize, create_worker_working_directory,
+    def test_initialize_worker(self, _delete_worker, initialize,
+                               create_worker_working_directory,
                                delete_worker_working_directory):
         """
         Assert that initialize_worker() calls Pulp's initialization code and the appropriate worker
-        monitoring code.
+        monitoring code for a non-resource mananger worker.
         """
         sender = mock.MagicMock()
```

Comment:
Is ``sender`` just a string in normal operation? If so, I suggest making it a string here for simplicity, and make it actually start with the value you want. It seems silly that it was ever a ``MagicMock`` to begin with.

Then you can test both branches of the ``if`` statement by doing two tests with two different values of ``sender``.